Rewrite "rock paper scissors" project

cakegod opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because


## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->
- Improve clarity, wording and punctuation of the "rock paper scissors"

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.
-->
Closes #XXXXX

## Additional Information
<!-- Any other information about this PR, such as a link to a Discord discussion. -->


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,46 @@
+    - **Tip**: Use the console to make sure this returns the expected output before moving to the next step!
+4. Write a new function called `playRound` that plays a single round of Rock Paper Scissors. This `playRound` function should have two parameters: `playerSelection` and `computerSelection`, both taking string arguments. Then, return a string that declares the winner of the round, like so: `"You Lose! Paper beats Rock"`
+    *   Make sure your `playerSelection` parameter is case-insensitive so that users can input `rock`, `ROCK`, `RocK`, or any other variation.
+   -  **Important note:** you want to `return` the results of `playRound` call, *not* `console.log` them. You're going to use what you `return` later on, so let's test this function by using console.log to see the results:
```
```suggestion
5. Let's test this function by using `console.log` to see the results
   - **Important note:** later on, you want to `return` the results of the `playRound` function, *not* `console.log` them. However, for now you will be using `console.log` to test the results of the function.
```
- keep the numbering but add a note since this is a step in the assignment still where the learner is expected to add `console.log`s
- Whatever we decide, let's add consistent backticks around `console.log` to be in line with the previous line 

I made some edits for this section has always felt kind of "off" to me. To me this sounds like: 
>*"We want to `return`, so let's go ahead and `console.log`"*

What do you think of these changes? They're not set in stone, just a draft change I made for you to judge.

cakegod replied:
I think it would be better to `return` the result right at the start, never `console.log`.

We can then use `console.log` on the function call to check its output as: 

`console.log(playRound(playerSelection, computerSelection))`

ManonLef replied:
You're absolutely right! I misinterpreted that wrongly all the time and hence why it always sounded "off" to me to mention we should return but then console.log. Now it makes sense, it only took me a year!

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,46 @@
+      ~~~
+
+5. Write a new function called `game` that plays a 5 round game, keeps the game's score, and reports a winner or loser at the end. Use the previous function *inside* this `game` function.
+    *   You have not officially learned how to "loop" over code to repeat function calls... If you already know about loops from somewhere else or feel like learning more, feel free to use them. If not, don't worry! Just call your `playRound` function 5 times in a row. In the next lesson, we will cover loops.
```
```suggestion
   -   You have not officially learned how to "loop" over code to repeat function calls. However, if you already know about loops from somewhere else or feel like learning more, feel free to use them. If not, don't worry! Just call your `playRound` function 5 times in a row. In the next lesson, we will cover loops.
```

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,46 @@
+5. Write a new function called `game` that plays a 5 round game, keeps the game's score, and reports a winner or loser at the end. Use the previous function *inside* this `game` function.
+    *   You have not officially learned how to "loop" over code to repeat function calls... If you already know about loops from somewhere else or feel like learning more, feel free to use them. If not, don't worry! Just call your `playRound` function 5 times in a row. In the next lesson, we will cover loops.
+    *   At this point, you should be using `console.log` to display the results of each round and the winner at the end.
+    *   Use the `prompt` function to get input from the user. [Read the docs here if you need to.](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt)
     *   Feel free to re-work your previous functions if you need to. Specifically, you might want to change the return value to something more useful.
-    *   Feel free to create more "helper" functions if you think it would be useful.
+    *   Feel free to create more "helper" functions if you think they would be useful.
```
- `-` for list items per style guide 
- perhaps a more descriptive links for the MDN documentation about `prompt`

cakegod replied:
Maybe?
>  To obtain the choice from the human user, use the `prompt` function. You can refer to the [official MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) to learn how to use the prompt function and its parameters.

ManonLef replied:
I think we should also add the chapter :) 
> To obtain the choice from the human player, use the prompt method. You can refer to the [prompt() method documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) to learn how to use the prompt method and its parameters.

Feel free to simplify the last part after the link to your liking. It's probably too much prompt repetition.

ManonLef reviewed (changes requested):
Thanks for revisiting this lesson. I think it could indeed do with a nice makeover, great work! 

I left a few comments. I hope I. caught it all but it was quite a lot to review. Let me know your thoughts!

cakegod commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,46 @@
-We're going to make a simple implementation of grade-school classic "rock paper scissors". If you don't know what that is check the [Wikipedia article](https://en.wikipedia.org/wiki/Rock%E2%80%93paper%E2%80%93scissors) or [this](https://www.wikihow.com/Play-Rock,-Paper,-Scissors) detailed step-by-step. For the moment we're just going to play the game from the browser console, but __we will revisi ...
+In this project, we'll create a simple implementation of the classic grade-school game "rock paper scissors". If you're unfamiliar with this game, check out this [Wikipedia article](https://en.wikipedia.org/wiki/Rock%E2%80%93paper%E2%80%93scissors) or this [detailed step-by-step guide](https://www.wikihow.com/Play-Rock,-Paper,-Scissors).
+
+We'll start by playing the game entirely from the browser console. Later on, **we'll revisit this project to add a Graphical User Interface (GUI) with buttons and text**. So don't forget to save the code on GitHub! Some student solutions below might already have a GUI, but we'll cover that in a later lesson. Let's get started with a few quick exercises before we start the project!
```
I'll see what I can do with the note box.

 I guess **push** would be the more appropriate word. I'm okay with either, though.

ManonLef replied:
I like `push` for I feel it's more common than `saving to GH`, Or perhaps we `store` remote. I'm not sure. I don't feel strongly about it and will leave it up to you :)

ManonLef commented:
Sidenote: You can leave the resolving up to the reviewer usually. That way they can check and resolve their requests for changes once they are committed. 

I'll check back tomorrow for it's late here. Thanks for taking this on!

ManonLef commented:
![Screenshot 2023-08-04 at 22 50 35](https://github.com/TheOdinProject/curriculum/assets/81025586/ac082f94-2109-49d5-a2f6-0e38c9deb106)

Please re-request a review once you're ready for it. Thank you!

cakegod commented:
I've made significant changes to the original, including the requested changes. I'm unsure about using tip note boxes on the assignment; tell me what you think.

ManonLef commented:
Thanks @cakegod !

Those changes are a lot more than expected. I can't review them today anymore. 
I don't think the tips should be in lesson note elements no. They drag too much attention in my opinion with the added color and I believe I liked the previous ones better. 

I'll review this lesson once I get some time. Thanks again!

cakegod commented:
Indeed, this is also my overall feeling. 

We could probably have a specific tip note-box-like for the assignment box. Adding tips requires us to add them as bullet points, which is odd since they are not strictly part of the assignment.

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+1. Create a blank HTML document with a script tag.
+   1. We recommend writing the code in an external JavaScript file and linking it to the HTML file. You play this game entirely on the console, so don't worry about writing anything else in the HTML file.
+1. Your game plays against the computer—create a function to `return` the computer's choice:
+   1. Create a new function called `getComputerChoice`.
+   1. Make `getComputerChoice` *randomly* `return` one of the following string values: `"Rock"`, `"Paper"` or `"Scissors"`.
+   1. Remember to test what your function returns with [browser's developer tools](https://www.theodinproject.com/lessons/foundations-javascript-developer-tools) or `console.log`.
```
```suggestion
   - Create a new function `getComputerChoice`.
   - Write the code to let `getComputerChoice` randomly `return` one of the following string values: "Rock", "Paper" or "Scissors".
   - Test that your function returns what you expect using `console.log` or [the browser developer tools](https://www.theodinproject.com/lessons/foundations-javascript-developer-tools) before you move on to the next step.
```

Good idea to put the dev tools here!

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+   1. Create a new function called `getComputerChoice`.
+   1. Make `getComputerChoice` *randomly* `return` one of the following string values: `"Rock"`, `"Paper"` or `"Scissors"`.
+   1. Remember to test what your function returns with [browser's developer tools](https://www.theodinproject.com/lessons/foundations-javascript-developer-tools) or `console.log`.
+1. Your game plays turn by turn—create a function to take the human and computer choices, then `return` the round winner:
```
```suggestion
1. Your game plays turn by turn. Write a function named `playRound`, which takes the `playerSelection` and `computerSelection` to play a single round, and returns a winner announcement. 
```

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+   1. Make `getComputerChoice` *randomly* `return` one of the following string values: `"Rock"`, `"Paper"` or `"Scissors"`.
+   1. Remember to test what your function returns with [browser's developer tools](https://www.theodinproject.com/lessons/foundations-javascript-developer-tools) or `console.log`.
+1. Your game plays turn by turn—create a function to take the human and computer choices, then `return` the round winner:
+   1. Create a new function called `playRound`.
+   1. Define two parameters on `playRound`: `playerSelection` and `computerSelection`. Use those two parameters to take the human and computer choices as arguments.
+   1. Make `playRound` `return` a string value representing the round winner, such as: `"You Lose! Paper beats Rock"`.
```
```suggestion
   - Create a new function called `playRound`.
   - Define two parameters on `playRound`: `playerSelection` and `computerSelection`. Use those two parameters to pass the human and computer choices as arguments.
   - Make your function's `playerSelection` parameter case-insensitive (so users can input "rock", "ROCK", "RocK" or any other variation).
   - Write the code for your `playRound` function to `return` a string value representing the round winner, such as: "You Lose! Paper beats Rock".
   - example code:
```

- reworded some and added the case-insensitive part of the assignment back in.
- the below code should fit below the last list item, so indentation needs to increase. Feel free to reword the last list item to make it clear this is code to test if the function works.

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
-    *   Feel free to create more "helper" functions if you think it would be useful.
+   const playerSelection = "rock";
+   const computerSelection = getComputerChoice();
+   console.log(playRound(playerSelection, computerSelection));
```
```suggestion
   console.log(playRound(playerSelection, computerSelection));
```

the whole code block should go below last list item or place it back into it's own numbered list step to test our one round.

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
-### Quick exercises before starting
+<div class="lesson-note">
+<h4>Student solutions</h4>
+Some students' solutions below include buttons, text, and other elements. Those elements form what is known as a Graphical User Interface (GUI). You'll add those elements—the GUI, in a later lesson. So remember to push your code to GitHub.
```
```suggestion
Some students' solutions below include buttons, text, and other elements. Those elements form what is known as a Graphical User Interface (GUI). You'll add those elements, the GUI, in a later lesson. So remember to push your code to GitHub.
```

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+   console.log(playRound(playerSelection, computerSelection));
+   ~~~
+
+1. Your game plays for 5 rounds, and reports the winner or loser after the 5 rounds—create a function to represent the game:
```
```suggestion
1. Your game plays for 5 rounds. Write a function called `game` which will keep score for 5 rounds, utilizing your 'playRound' function, before it declares a winner:
```

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+   ~~~
+
+1. Your game plays for 5 rounds, and reports the winner or loser after the 5 rounds—create a function to represent the game:
+   1. Create a new function called `game`.
+   1. Get the human player choice using `prompt` function. To learn about `prompt`, check the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt)
+   1. Put the previous function—`playRound`—inside `game`.
+   1. Play 5 rounds by calling `playRound` 5 times.
+   1. If you know about loops, use them. If not, don't worry! We cover loops in the next lesson. 
```
```suggestion
   - Create a new function called `game`.
   - Get the human player choice using the `prompt` method. Read [this MDN article about prompts](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) if you need to learn more about it.  
   - Put the previous `playRound` function inside `game`.
   - Play 5 rounds by calling `playRound` 5 times.
   - Re-work your previous functions or create more helper functions if you feel the need to. Specifically, you might want to change the return values to something more useful.
   - If you already know about loops, you can use them. If not, don't worry! We cover loops in the next lesson. 
```
- descriptive/accessible link 
- important hint about the return values added back in (needed for score and winner tracking)

ManonLef reviewed (changes requested):
That was a lot to review! 

I do think this lesson was in need of some clarification judging by the amount of questions surrounding the wording in our discord. 

For big changes or full lesson rewrites like this we usually start with an issue first so discussion can happen there to flesh out the exact details regarding areas of improvement. 
I expect most of my comments could use some discussing as well, and I welcome your input. 

I also would like another maintainer to review this after we have gone through most since it encompasses a full rewrite at this point. 

I may have missed things so please go through it carefully but my key points are this:
- the different -- lines between sentences are not what we normally use so I removed them
- code elements reserved for code and not for boldening or other emphasis
- the numbered list should be fully encompassing what the step is. I approached it as: could someone experienced code this without the added steps (not numbered) below?
- the steps below are breaking the problem in the numbered list item down.

Let me know what you think!

ManonLef commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
 ### Introduction
 
-We're going to make a simple implementation of grade-school classic "rock paper scissors". If you don't know what that is check the [Wikipedia article](https://en.wikipedia.org/wiki/Rock%E2%80%93paper%E2%80%93scissors) or [this](https://www.wikihow.com/Play-Rock,-Paper,-Scissors) detailed step-by-step. For the moment we're just going to play the game from the browser console, but __we will revisi ...
+For this project, you'll create a console-only version of the grade-school game: **Rock Paper Scissors**. To learn more, or if you're unfamiliar with the game, read this [Wikipedia article](https://en.wikipedia.org/wiki/Rock%E2%80%93paper%E2%80%93scissors) or this [detailed step-by-step guide](https://www.wikihow.com/Play-Rock,-Paper,-Scissors)
```
looking at this again, we should probably decide between "Rock Paper Scissors" or "rock paper scissors" and stay consistent 😁

cakegod replied:
I prefer the former.

For the changes, I prefer this concise version: 

> For this project, you'll create the grade-school game: **Rock Paper Scissors**, played entirely on the console. To learn more, or if you're unfamiliar with the game, read this [Wikipedia article about the rock paper scissors game]

cakegod commented:
> That was a lot to review!
> 
> I do think this lesson was in need of some clarification judging by the amount of questions surrounding the wording in our discord.
> 
> For big changes or full lesson rewrites like this we usually start with an issue first so discussion can happen there to flesh out the exact details regarding areas of improvement. I expect most of my comments could use some discussing as well, and I welcome your input.
> 
> I also would like another maintainer to review this after we have gone through most since it encompasses a full rewrite at this point.
> 
> I may have missed things so please go through it carefully but my key points are this:
> 
> * the different -- lines between sentences are not what we normally use so I removed them
> * code elements reserved for code and not for boldening or other emphasis
> * the numbered list should be fully encompassing what the step is. I approached it as: could someone experienced code this without the added steps (not numbered) below?
> * the steps below are breaking the problem in the numbered list item down.
> 
> Let me know what you think!

The initial idea was to fix punctuation and make minor improvements. It then evolved into something bigger, sorry!

I like the em dashes; they emphasize certain parts. I used them since I saw nothing in the layout style guide forbidding their use.

In a way, the multiple steps are mixed into each step in the current assignment, so I split them into sub-steps.

cakegod commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+
+### Quick exercise before starting
+1. Identify three ways to include JavaScript in a page.
+2. Check if the page includes JavaScript:
+   - Write `console.log("Hello World");` in JavaScript.
+   - Check if `console.log("Hello World");` logs in the browser's console.
```
The duplication of the console logs gave me trouble; I didn't know what to do with them! :laughing:

ManonLef commented:
> The initial idea was to fix punctuation and make minor improvements. It then evolved into something bigger, sorry!

It happens! It just takes a lot more resources than the initial PR you filed. I do think this project could be enhanced and think you put in a great step to doing so!

> I like the em dashes; they emphasize certain parts. I used them since I saw nothing in the layout style guide forbidding their use.

They are not mentioned but also stand out for they are not generally used and I think the sentences don't lose a lot by not inserting them here. 

> In a way, the multiple steps are mixed into each step in the current assignment, so I split them into sub-steps.

Yes I can see that, and I left the sub-steps. I do think the numbering needs to go for those. And I rewrote the main steps a bit. The substeps are still an elaboration for those who need it. The only thing I wondered while reviewing is if it now doesn't go overboard on handholding, but judging from the amount of issues people have with this project being the first JS one, I think it will be a hard project either way.

cakegod commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+### Problem solving approach
 
-Finally, this is your first JavaScript program built from scratch, so don't forget the previous lesson on problem solving. Plan your solution out before writing any code, and test each piece as you build to ensure it is working before moving on to the next!
+Remember the [previous lesson on problem solving](https://www.theodinproject.com/lessons/foundations-problem-solving) before diving into the project. Plan your solution—write the code—and then ensure the code works.
```
I am not fond of the redundancy of:
>Since this is your first JavaScript program built from scratch

I believe it serves no purpose other than adding more text to read. The learner already knows that this is their first program (if it is). There is no point in telling them.

cakegod commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,54 @@
+
+1. Create a new Git repository for your project.
+1. Create a blank HTML document with a script tag.
+   1. We recommend writing the code in an external JavaScript file and linking it to the HTML file. You play this game entirely on the console, so don't worry about writing anything else in the HTML file.
```
Great point about the link!

cakegod commented:
> Yes I can see that, and I left the sub-steps. I do think the numbering might need to go for those. And I rewrote the main steps a bit. The substeps are still an elaboration for those who need it. The only thing I wondered while reviewing is if it now doesn't go overboard on handholding, but judging from the amount of issues people have with this project being the first JS one, I think it will be a hard project either way.

The current assignment already handholds. We're simply making the assignment clearer by breaking the steps into smaller parts. :wink: 

I'll commit your changes and re-check everything.

ManonLef commented:
> The current assignment already handholds. We're simply making the assignment clearer by breaking the steps into sub-steps. 😉

I can't say I disagree, it's just more obfuscated by it all being crammed in a sentence I guess 😅

CBID2 commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,58 @@
-3.  Your game is going to play against the computer, so begin with a function called `getComputerChoice` that will randomly return either 'Rock', 'Paper' or 'Scissors'.  We'll use this function in the game to make the computer's play. *Tip: use the console to make sure this is returning the expected output before moving to the next step!*
-4.  Write a function that plays a single round of Rock Paper Scissors.  The function should take two parameters - the `playerSelection` and `computerSelection` - and then return a string that declares the winner of the round like so: `"You Lose! Paper beats Rock"`
-    *   Make your function's playerSelection parameter case-insensitive (so users can input `rock`, `ROCK`, `RocK` or any other variation).
+Since this is your first JavaScript program built from scratch, remember the [previous wise words from the problem-solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving). Before diving into this project, first plan or pseudocode your solution, then write the code, then ensure the code works.
```
[Proofreading] The last sentence seems to describe some steps, so I provided a suggestion on how to make it clearer. 
```suggestion
Since this is your first JavaScript program built from scratch, remember the [previous wise words from the problem-solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving). Before diving into this project, First, plan or pseudocode your solution. Next, write the code. Lastly, ensure the code works.
```

ManonLef replied:
Thank you, we are still in the exploratory phase it seems so no final version to review yet.

CBID2 commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,58 @@
+   console.log(playRound(playerSelection, computerSelection));
+   ~~~
+
+1. Your game plays for 5 rounds. Write a function called `game` which will keep score for 5 rounds, utilizing your 'playRound' function, before it declares a winner:
```
[Proofreading] Parts of this step were a bit unclear, so I gave some ideas on how to revise it. Check out my suggestions below:
```suggestion
1. Write a function called `game` which will keep score for 5 rounds and utilize your 'playRound' function before it declares a winner:
```

ManonLef replied:
I hope you saw my last comment. We're still finding a direction for this so no need to review just yet. Thank you however :)

CBID2 reviewed (commented):
Hi @cakegod! :) I found some parts of your pull request that could be improved, so check out my in-line comments for context. Other than that, your entry is off to a great start! :) Great job! :)

ManonLef commented:
@cakegod I have converted this lesson to draft for now so people can see it's still a work in progress.

cakegod commented:
I have new changes, but they are... abundant. I will fix the merge and commit the existing changes before opening an issue or new PR.

ManonLef commented:
@cakegod what's the status on this? If you're ready for a review again after making changes, there's a button to request it. 
![Screenshot 2023-10-03 at 11 53 01](https://github.com/TheOdinProject/curriculum/assets/81025586/ab426721-28d2-45dc-9755-e03445f491e4)

cakegod commented:
> @cakegod what's the status on this? If you're ready for a review again after making changes, there's a button to request it. 
> ![Screenshot 2023-10-03 at 11 53 01](https://github.com/TheOdinProject/curriculum/assets/81025586/ab426721-28d2-45dc-9755-e03445f491e4)
> 

Still a work in progress. I have started creating a checklist of pain points, but I have yet to write fixes for them. I'll see if I can finish it this week.

ManonLef commented:
> > @cakegod what's the status on this? If you're ready for a review again after making changes, there's a button to request it.
> > ![Screenshot 2023-10-03 at 11 53 01](https://user-images.githubusercontent.com/81025586/272220939-ab426721-28d2-45dc-9755-e03445f491e4.png)
> 
> Still a work in progress. I have started creating a checklist of pain points, but I have yet to write fixes for them. I'll see if I can finish it this week.

Thanks for the update. There's no rush whatsoever! I just wanted to make sure you weren't waiting for a review or anything.

cakegod commented:
It's time to finish this before I forget again. Tell me what you think about the latest changes. I'm not sure if we should add more tips or leave it as it is.

ManonLef commented:
I wanted to let you know it will be a while before I get to sit down for this. My focus is on an internship currently and I expect to get back to maintaining at the start of the new year. Thanks for the work you've put in!

kamranzafar4343 reviewed (commented):
Just reviewing it, Thank you all for your contribution! I appreciate the effort you've put into this commit. The code looks clean and well-organized. Keep up the good work!

nikitarevenco commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,81 @@
+
+It's best practice to link to an external JavaScript file inside this script tag. Using an external JavaScript file keeps your HTML file clean and organized.
+
+You don't have to write additional code in the HTML file. This game is played entirely on the console.
```
```suggestion
You don't have to write additional code in the HTML file. This game is played entirely in the console.
```

nikitarevenco commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,81 @@
+
+#### Step 2: get computer choice function
+
+Your game will be played against the computer. Create a function named `getComputerChoice` that randomly returns 'Rock', 'Paper' or 'Scissors':
```
```suggestion
Your game will be played against the computer. You will make a function that randomly returns 'Rock', 'Paper' or 'Scissors':
```
Removes unnecessary repetition.
`Create a function named getComputerChoice`

The instructions are already well-written in the bullet point list

nikitarevenco commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,81 @@
+
+1. Create a new function named `getComputerChoice`.
+1. Write the code to let `getComputerChoice` randomly `return` one of the following string values: "Rock", "Paper" or "Scissors". 
+   - **Hint**: The [Math.random](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) method returns a random number that's greater than or equal to 0 and less than 1. Think about how you can use this to conditionally return one of the multiple choices.
```
```suggestion
   - **Hint**: The [`Math.random`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) method returns a random number that's greater than or equal to 0 and less than 1. Think about how you can use this to conditionally return one of the multiple choices. You may find the [`Math.floor`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/floor) method useful.
```

It could potentially be confusing if the learner doesn't know about the Math.floor method if they wanted to generate a random index.

cakegod replied:
Technically, they shouldn't be using arrays as they haven't learned about it.

nikitarevenco replied:
> Technically, they shouldn't be using arrays as they haven't learned about them.

Oh yeah, on point. No need to add that link here then

nikitarevenco commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,42 +1,81 @@
+
+#### Step 3: play round function
+
+Your game will be played round by round. Write a function named `playRound` that takes the human and computer player choices as arguments, plays a single round and returns a winner announcement:
```
```suggestion
Your game will be played round by round. You will make a function that takes the human and computer player choices as arguments, plays a single round and returns a winner announcement:
```

Again, there's no point saying the instructions twice

nikitarevenco reviewed (commented):
I really like this rewrite, I've read and I have some small ideas that could improve the readability

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
 ### Introduction
 
-We're going to make an implementation of grade-school classic "rock paper scissors". If you don't know what that is check the [Wikipedia article](https://en.wikipedia.org/wiki/Rock%E2%80%93paper%E2%80%93scissors) or [this](https://www.wikihow.com/Play-Rock,-Paper,-Scissors) detailed step-by-step. For the moment we're just going to play the game from the browser console, but **we will revisit this ...
+For this project, you will create the game **Rock Paper Scissors**. This game will be played entirely on the console. To learn more, or if you're not familiar with the game, read this [Wikipedia article](https://en.wikipedia.org/wiki/Rock%E2%80%93paper%E2%80%93scissors) or this [detailed step-by-step guide.](https://www.wikihow.com/Play-Rock,-Paper,-Scissors)
```
```suggestion
For this project, you will create the game [Rock Paper Scissors](https://www.wikihow.com/Play-Rock,-Paper,-Scissors). This game will be played entirely in the console.
```
Really don't think we need 2 separate links to explain RPS and its rules. This can be shortened as such, with the new link text being sufficiently descriptive (was planned as part of descriptive link changes already).

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
 
-### Quick exercises before starting
+<div class="lesson-note">
+Some of the student solutions below contain buttons, text, and other elements. These elements are part of what is called a graphical user interface (GUI) You'll create the GUI in a later lesson. In the meantime, remember to commit your code to GitHub.
```
```suggestion
Some of the student solutions below contain buttons, text, and other elements. These elements are part of what is called a graphical user interface (GUI). You'll create the GUI in a later lesson. In the meantime, remember to commit your code to GitHub.
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+Some of the student solutions below contain buttons, text, and other elements. These elements are part of what is called a graphical user interface (GUI) You'll create the GUI in a later lesson. In the meantime, remember to commit your code to GitHub.
+</div>
+
+### Quick exercise before starting
+
+1. Identify three ways to include JavaScript in a webpage.
+1. Check if the webpage includes JavaScript:
+   - Write `console.log("Hello World")` in JavaScript.
+   - Check if "Hello World" is logged in the browser console once you open your webpage.
```
This step feels a little out of place to me (the original too).
We have not yet instructed the learner to set anything up, only identify the 3 ways to include JS in a web page. Suddenly we're now saying "write JS and check if the web page runs it" - but we haven't made a web page yet.

I feel like the "identify three ways to include JS in a web page" bit might be better placed at the start of the assignment section itself, though I don't feel too strongly about this.

I do think the "test with a console log" but should at the very least come after we've instructed the learner to create a blank HTML with script tag. Though I wonder by this point, is it a necessary step? I don't feel strongly about its inclusion/exclusion but do thing it makes way more practical sense there and not before.

cakegod replied:
Good point.
 
We can add the console log part as a third step in the initial structure step. I don't think "Identify three ways to include JavaScript in a webpage." step is necessary.

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+### Problem solving approach
 
-Finally, this is your first JavaScript program built from scratch, so don't forget the previous lesson on problem solving. Plan your solution out before writing any code, and test each piece as you build to ensure it is working before moving on to the next!
+Remember the [previous wise words from the Problem Solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving) because this is your first JavaScript program built from scratch. Before you start this project, follow these steps for each step of the project:
```
```suggestion
Remember the previous wise words from the [Problem Solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving) because this is your first JavaScript program built from scratch. Before you start this project, follow these steps for each step of the project:
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
-Don't forget to commit early & often! You can [reference the Commit Message lesson here](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/commit-messages)!
+Remember to commit early and often! To refresh your memory, check out the [commit messages lesson](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/commit-messages).
+
+#### Step 1: initial structure
```
```suggestion
#### Step 1: Initial structure
```
Sentence case requires [capitalising the first word after a colon in a heading](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case).

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+
+You don't have to write additional code in the HTML file. This game is played entirely via the console.
+
+#### Step 2: get computer choice function
```
```suggestion
#### Step 2: Get computer choice function
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+   - **Hint**: The [Math.random](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) method returns a random number that's greater than or equal to 0 and less than 1. Think about how you can use this to conditionally return one of the multiple choices.
+1. Test that your function returns what you expect using `console.log` or [the browser developer tools](https://www.theodinproject.com/lessons/foundations-javascript-developer-tools) before advancing to the next step.
+
+#### Step 3: play round function
```
```suggestion
#### Step 3: Play round function
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+Your game will be played round by round. You will make a function that takes the human and computer player choices as arguments, plays a single round and returns a winner announcement:
+
+1. Create a new function named `playRound`.
+1. Define two parameters on `playRound`: `playerChoice` and `computerChoice`. Use those two parameters to take the human and computer choices as arguments.
```
```suggestion
1. Define two parameters for `playRound`: `playerChoice` and `computerChoice`. Use these two parameters to take the human and computer choices as arguments.
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+1. Create a new function named `playRound`.
+1. Define two parameters on `playRound`: `playerChoice` and `computerChoice`. Use those two parameters to take the human and computer choices as arguments.
+1. Make your function's `playerSelection` parameter case-insensitive so that players can input "rock", "ROCK", "RocK", or other variations.
+1. Write the code for your `playRound` function to `return` a string value representing the round winner, such as: "You Lose! Paper beats Rock".
```
```suggestion
1. Write the code for your `playRound` function to `return` a string value representing the round winner, such as: "You lose! Paper beats Rock".
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
````diff
@@ -1,44 +1,84 @@
+```
 
-   - Make your function's playerSelection parameter case-insensitive (so users can input `rock`, `ROCK`, `RocK` or any other variation).
+#### Step 4: game function
````
```suggestion
#### Step 4: Play game function
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
+#### Step 4: game function
 
-1. **Important note:** you want to `return` the results of this function call, *not* `console.log()` them. You're going to use what you `return` later on, so let's test this function by using console.log to see the results:
+Your game will play 5 rounds. Write a function named `game` that uses `playRound` to play 5 rounds, keeps score, and declares a winner at the end:
```
```suggestion
Your game will play 5 rounds. Write a function named `playGame` that uses `playRound` to play 5 rounds, keeps score, and declares a winner at the end:
```
I think we should keep consistent with the way we name the functions. `playGame` makes more sense.

cakegod replied:
Can't believe I missed the prefix. :eyes:

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
-   function playRound(playerSelection, computerSelection) {
-     // your code here!
-   }
+1. Create a new function named `game`.
+1. Get the human player choice using the `prompt` method. Read this [MDN article about prompt method](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) learn more about it.
+1. Put the previous `playRound` function inside `game`.
```
```suggestion
1. Create a new function named `playGame`.
1. Get the human player choice using the `prompt` method. Read this [MDN article about the prompt method](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) learn more about it.
1. Put the previous `playRound` function inside `playGame`.
```

MaoShizhong commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,84 @@
-   - Use `prompt()` to get input from the user. [Read the docs here if you need to.](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt)
-   - Feel free to re-work your previous functions if you need to. Specifically, you might want to change the return value to something more useful.
-   - Feel free to create more "helper" functions if you think it would be useful.
+If you already know about loops, you can use them. If not, don't worry! Loops will be covered in the next lesson.
```
Not sure if it's worth mentioning this? Not a strong feeling, just wondering if there's really a point to mention it.

MaoShizhong reviewed (commented):
Really like the overall direction this is going in @cakegod!
A couple of suggestions and considerations below, as well as some small fixes.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
 
-1. Identify three ways to include JavaScript in a page.
-1. Test it out! Write `console.log("Hello World");` in JavaScript and check to see if it displays in the browser's console.
+Remember the previous wise words from the [Problem Solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving) because this is your first JavaScript program built from scratch. Before you start this project, follow these steps for each step of the project:
```
More of a nit for how the current verbiage sounds:

```suggestion
Since this is the first JavaScript project being built from scratch, it's important to remember the wise words from the [Problem Solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving). Before you start this project, follow these steps for each step of the project:
```

cakegod replied:
I like it.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+Remember the previous wise words from the [Problem Solving lesson](https://www.theodinproject.com/lessons/foundations-problem-solving) because this is your first JavaScript program built from scratch. Before you start this project, follow these steps for each step of the project:
 
-Finally, this is your first JavaScript program built from scratch, so don't forget the previous lesson on problem solving. Plan your solution out before writing any code, and test each piece as you build to ensure it is working before moving on to the next!
+1. Plan or pseudocode your solution.
+1. Write the code.
+1. Test your code to make sure it works.
```
If this is meant to echo the problem solving lesson, we probably want these items to be about understanding the problem, planning, writing pseudo code, and then dividing and conquering, no?

Maybe it's the wording leading into it, though. "Before you start this project" could maybe be removed, then just say, "For each step in this project, be sure to do the following:"

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+
+You don't have to write additional code in the HTML file. This game is played entirely via the console.
+
+#### Step 2: Get computer choice function
```
Nit not a blocker, but maybe "Logic for getting the computer choice". Same would apply to the other sections e.g. "Logic for playing a round" etc. Fine keeping it as-is, too, so up to you.

cakegod replied:
I'm fine with a more detailed heading.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
-Don't forget to commit early & often! You can [reference the Commit Message lesson here](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/commit-messages)!
+Remember to commit early and often! To refresh your memory, check out the [commit messages lesson](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/commit-messages).
+
+#### Step 1: Initial structure
```
I do kind of like how we split projects up into more easily consumable pieces like this. One or two HTML/CSS projects in Foundations do something similar, though those use "Iteration 1/2/3/etc". I don't really have a strong preference either way, but we should decide on a consistent verbiage sooner than later (doesn't have to be in this PR)

cakegod replied:
I'm personally not a fan of "iteration"; it doesn't sound very fitting. We'll discuss it later in a different PR.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+
+#### Step 2: Get computer choice function
+
+Your game will be played against the computer. You will make a function that randomly returns 'Rock', 'Paper' or 'Scissors':
+
+1. Create a new function named `getComputerChoice`.
+1. Write the code so that `getComputerChoice` randomly `return` one of the following string values: "Rock", "Paper" or "Scissors".
```
Some small tweaks

```suggestion
Your game will be played against the computer. You will write a function that randomly returns "Rock", "Paper" or "Scissors".

1. Create a new function named `getComputerChoice`.
1. Write the code so that `getComputerChoice` will randomly `return` one of the following string values: "Rock", "Paper" or "Scissors".
```

Probably not a bad idea to provide function names for users for the first project in Foundations, to get them used to the idea of clean code better.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+
+1. Create a new function named `getComputerChoice`.
+1. Write the code so that `getComputerChoice` randomly `return` one of the following string values: "Rock", "Paper" or "Scissors".
+   - **Hint**: The [Math.random](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random) method returns a random number that's greater than or equal to 0 and less than 1. Think about how you can use this to conditionally return one of the multiple choices.
```
Kind of feel like maybe this would be worth omitting, and just telling users to randomly return a string (and just say "This would be a great time to Google how to do so!" or something). That said, if this has been a pain point for users on Discord for this project I'm fine leaving this here.

MaoShizhong replied:
It has actually been somewhat of a pain point, surprisingly enough.
It's been somewhat often where just mentioning `Math.random` gets people on the right tracks but despite a brief mention of it earlier in the curriculum, a lot of people seem to be stuck at what to even think about looking into here.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+
+#### Step 3: Play round function
+
+Your game will be played round by round. You will make a function that takes the human and computer player choices as arguments, plays a single round and returns a winner announcement:
```
```suggestion
Your game will be played round by round. You will write a function that takes the human and computer player choices as arguments, plays a single round and returns a winner announcement.
```

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+#### Step 4: Play game function
 
-1. **Important note:** you want to `return` the results of this function call, *not* `console.log()` them. You're going to use what you `return` later on, so let's test this function by using console.log to see the results:
+Your game will play 5 rounds. Write a function named `playGame` that uses `playRound` to play 5 rounds, keeps score, and declares a winner at the end:
```
```suggestion
Your game will play 5 rounds. You will write a function named `playGame` that calls `playRound` to play 5 rounds, keeps score, and declares a winner at the end.
```

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
-     // your code here!
-   }
+1. Create a new function named `playGame`.
+1. Get the human player choice using the `prompt` method. Read this [MDN article about the prompt method](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) learn more about it.
```
Could also combine the link with the prompt method text:

```suggestion
1. Get the human player choice using the [`prompt` method](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt).
```

cakegod replied:
I'm not sure about combining both as it looks are odd:
![image](https://github.com/TheOdinProject/curriculum/assets/36206278/bdf755c4-438a-4afd-8c04-c28a830a52b5)

MaoShizhong replied:
Agreed that visually, at least for 2 word links (1 in backticks, 1 not), it really is odd to look at, and can easily be misinterpreted at 2 separate links at a glance.
At least with more words, the ambiguity is less.

thatblindgeye replied:
Good call, I'm fine with it as-is then

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
-   }
+1. Create a new function named `playGame`.
+1. Get the human player choice using the `prompt` method. Read this [MDN article about the prompt method](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) learn more about it.
+1. Put the previous `playRound` function inside `playGame`.
```
Is the intent with this line here that users should literally move their playRound declaration inside of playGame, or just that they should call playRound inside of playGame? The wording makes it sound like the former which we probably don't want users to do. We could honestly probably just remove this step since we're already telling users to call playRound 5x on the next line.

cakegod replied:
This depends on where we want the user to declare the scores. If the scores are declared inside the playGame, we won't be able to increment the scores inside the playRound. We can use the return of playRound to conditionally increment the winner's score variable declared in playGame, but this makes the project much more complicated.

So, I thought about those two simple ending outcomes we want to guide the Odinites to: https://gist.github.com/cakegod/2f5f63295ff606a1a729f67564cd7d43.

thatblindgeye replied:
Another option would be declaring the scores as global variables, though yeah as the project is currently written and remains written here the intent seems to be that playGame houses the scores.

If you feel having the playRound moved inside of playGame is the better option I'm good with that. Maybe just tweak the verbiage of this instruction to, "Move your `playRound` function so that it's declared inside of the new `playGame` function"

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,80 @@
+1. Get the human player choice using the `prompt` method. Read this [MDN article about the prompt method](https://developer.mozilla.org/en-US/docs/Web/API/Window/prompt) learn more about it.
+1. Put the previous `playRound` function inside `playGame`.
+1. Play 5 rounds by calling `playRound` 5 times.
+   - **Hint**: When you assign a function call to a variable, the return value of that function is assigned to the variable. Accessing the variable afterward will only provide the assigned value; it doesn't recall the function. You need to recall the choice functions to get new choices for each round.
```
If we're keeping the verbiage about loops it would make more sense to include it here

```suggestion
   - **Hint**: When you assign a function call to a variable, the return value of that function is assigned to the variable. Accessing the variable afterward will only provide the assigned value; it doesn't recall the function. You need to recall the choice functions to get new choices for each round.
   - If you already know about loops, you can use them. If not, don't worry! Loops will be covered in the next lesson.
```

thatblindgeye reviewed (commented):
Some additional comments below. Some are nits/things I wouldn't block over so lemme know if you have thoughts on them.

thatblindgeye commented on foundations/javascript_basics/project_rock_paper_scissors.md:
```diff
@@ -1,44 +1,78 @@
-   function playRound(playerSelection, computerSelection) {
-     // your code here!
-   }
+#### Step 4: Write the logic to entire the play the game
```
```suggestion
#### Step 4: Write the logic to play the entire game
```

thatblindgeye reviewed (changes requested):
One more tweak below. Would you mind also looking at the two unresolved comments from me above?

thatblindgeye reviewed (approved):
Other than the above, looks good. Awesome job on this!

cakegod commented:
I am not sure if we want to include validation on the human player input, as it would require a loop or recursion. :thinking:

MaoShizhong commented:
> I am not sure if we want to include validation on the human player input, as it would require a loop or recursion. 🤔

I'd say no need for validation for that exact reason.
